internal/prompt/prompt.go

```diff
@@ -158,7 +177,15 @@ func (p *Prompt) Select(title, message string, choices []string, defaultChoice *
 }
 
 // Confirm prompts user for yes or no response.
-func (p *Prompt) Confirm(title, message string, defaultChoice *bool) (bool, error) {
+func (p *Prompt) Confirm(title, message string, defaultChoice *bool, forcedChoice *bool) (bool, error) {
+	if p.isForced {
+		if forcedChoice == nil {
+			return false, errs.New("No force option given for force-enabled prompt")
+		}
+		logging.Debug("Prompt %s confirmed with choice %v in force mode", title, forcedChoice)
+		return *forcedChoice, nil
+	}
```

Comment:
Do we even need to check `isForced` here? I'm just not seeing why we have to set the prompt to a forced mode in order to use the forced choice. It's a bit different than the interactive use case.

Reply:
We do, because we are always supplying a forced choice for the prompts that support it. Just because we're supplying the choice doesn't mean force is enabled, so we do need to check.

When we initialize a prompt, we're supplying the `isInteractive` field inheritently, which depends on the terminal or if the `--non-interactive` flag is passed. That's the only thing different with the interactive/non-interactive case; we're still checking for `isInteractive` in this method.

Note: We cannot initialize prompt with a `isForced` because that information is not yet known, hence the need to set forced mode later.

I hope this makes sense.

Reply:
It does, but I still think we could get away without having an `isForced` property here. We pass nil to the `check` function if we're not forcing a choice otherwise it's explicit. This feels like we're forcing (no pun intended) this forced state to follow the same pattern as the interactive state when they are different things.

Anyways that's just my 2 cents. I'm fine with whatever way you want to do this. 👍